### PR TITLE
LPS-113333 Improve YAML errors detection to provide better output to REST Builder users

### PR DESCRIPTION
**Pull request by javierdearcos**

### CODE REVIEW

**Comment by javierdearcos** [comment]
@nhpatt

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLFileException.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			markedYAMLException);
+	}
+
+	private static String _getProblem(MarkedYAMLException markedYAMLException) {
+		if (markedYAMLException.getCause() instanceof MarkedYAMLException) {
+			return _getProblem(
+				(MarkedYAMLException)markedYAMLException.getCause());
+		}
+
+		Optional<Mark> optionalMark = Optional.ofNullable(
```
Comment (nhpatt) [inline modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLFileException.java:43]: should be markOptional

Comment (javierdearcos) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/yaml/YAMLFileException.java]: updated

**Comment by nhpatt** [comment]
LGTM :)

**Comment by nhpatt** [comment]
ci:forward

**Pull request closed without merging**